### ISSUE TITLE
Help in writing zone

### ISSUE THREAD

**Issue by kesavkolla** [zone-eu/zone-mta#236]
I need help in writing zones.

I've the following requirement:
all default outbound emails should go through sendgrid (MTA)

all internal emilas to my own domain (fhirhub.io) should be handled via haraka

all emails from a given user should go through G-Suite SMTP.

I could write default zone but I'm not able to understand how to write zone file for other two scenarios.

Appreciate some help.

**Comment by louis-lau** [comment]
I would help to know why you're having trouble, these sound like simple scenario's which can be found in the readme.

**Comment by kesavkolla** [comment]
If I add host, port in default zone and if I don't add any host port in other zone what's the behavior?

How to specify sender in zone?  I only see recipientDomains or senderDomains in the document.

**Comment by louis-lau** [comment]
Not sure what you mean by the first question. The behavior is whatever you specify in the zone. If you don't specify a port and host it won't send it to that port and host for that zone.

You can route base on a header value:
https://github.com/zone-eu/zone-mta#routing-based-on-specific-header-value

This could be the `from` header for example.

**Comment by kesavkolla** [comment]
I mean is there any way to specify use MX instead of MTA?

**Comment by louis-lau** [comment]
That's the default behaviour. Add an ip pool and it will send to the appropriate MX.

**Comment by kesavkolla** [comment]
Also is there way to say use STARTTLS?

**Comment by louis-lau** [comment]
It will use STARTTLS by default, then fallback to unencrypted if the other server doesn't support it.

**Comment by kesavkolla** [comment]
Thanks @louis-lau for all the replies.

**Comment by kesavkolla** [comment]
I've the following in a file **zones/developer.toml**

```toml
[developer]
preferIPv6=false
ignoreIPv6=true
processes=4
connections=10
pool="default"

# If you want to send messages for this zone to next MTA instead of actual MX, then
# uncomment following lines and change values
host = "smtp-relay.gmail.com"
port = 587
[developer.auth]
user = "myuser@mydomain"
pass = "mypassword"

[developer.routingHeaders]
From = "developer@hotelsoft.io"
```

Still the message is routing via default route.  This zone is not picked up.

**Comment by louis-lau** [comment]
Sorry, the from header is actually with a capital F, and is formatted like this:
```
From: Kesav Kolla <notifications@github.com>
```

I suggest you check the headers on a sent message to find out. It's also there's another header like `X-AuthUser` or something that you can use more reliably, but that completely depends on your setup.

**Comment by kesavkolla** [comment]
I added a log message in `lib/sending-zone.js`

```js
module.exports.findByHeaders = headers => {
    console.log(routingHeaders.size, routingHeaders);
    console.log(headers);


```

what I observe is the routingHeaders is empty.  In the same file I added log lines in **init** function there I do see routingHeaders map is getting populated.  Something is missing in between.

I don't see `X-AuthUser` header when I print the headers.  Yes I would love to use auth user instead of from address.
